src/plugin.ts

```diff
@@ -0,0 +1,31 @@
+import { Context } from "./types/context";
+import { userStartStop } from "./handlers/user-start-stop";
+import { Env, PluginInputs } from "./types";
+
+import { Octokit } from "@octokit/rest";
+import { createClient } from "@supabase/supabase-js";
+import { createAdapters } from "./adapters";
+import { Database } from "./adapters/supabase/types/database";
+import { PrettyLogs } from "./adapters/supabase/pretty-logs";
+
+export async function startStopBounty(inputs: PluginInputs, env: Env) {
+  const octokit = new Octokit({ auth: inputs.authToken });
+  const supabase = createClient<Database>(env.SUPABASE_URL, env.SUPABASE_KEY);
+
+  const context: Context = {
+    eventName: inputs.eventName,
+    payload: inputs.eventPayload,
+    config: inputs.settings,
+    octokit,
+    env,
+    logger: new PrettyLogs(),
+    adapters: {} as ReturnType<typeof createAdapters>,
```

Comment:
```suggestion
    adapters: createAdapters(supabase, context);
```

Doesn't this work?

Reply:
No because `context` is not instantiated at this point and you cannot use a `this` context either